Update Claim a Facility claim form & profile fields

kellyi opened the pull request:
## Overview

- add job_title and linkedin_profile fields to claim a facility claim form
- enable claimants to control visibility of facility website
- split facility_name field into facility_name_english and
facility_name_non_english profile fields
- add facility_workers_count profile field
- add facility_female_workers_percentage profile field
- add facility_type field & choices for facility claim profile
- add other_facility_type field for facility claim profile
- adjust display order of claim info fields on facility details screen

Connects #617 
Connects #634 

## Demo

![Screen Shot 2019-07-08 at 10 29 30 PM](https://user-images.githubusercontent.com/4165523/60854942-eb548d80-a1cf-11e9-83af-3534cf582e4f.png)

![Screen Shot 2019-07-08 at 10 29 50 PM](https://user-images.githubusercontent.com/4165523/60854948-ee4f7e00-a1cf-11e9-9ade-6e21bea50e48.png)

## Notes

There are a bunch of migrations here but I'll squash them when the other PRs are merged in -- so I didn't bother renaming anything.

I made discrete followup issues for handling

- product type #651
- production type #652
- certifications #653
- membership bodies #654

## Testing Instructions

- get this branch, then resetdb & processfixtures
- sign in and turn on claim a facility
- try to claim a facility and verify that you see the new claim form fields, with validation
- sign in as an admin and view the claim, verifying that you see the new claim form fields displayed in the dashboard
- approve the claim, then sign back in as the user who has claimed the facility and visit the claim edit profile page
- verify that you can update and save data in the new inputs
- visit the facility details page and verify that the new fields are visible (if they aren't null) and that the order matches the order requested in #617 

## Checklist

- [ ] `fixup!` commits have been squashed
- [ ] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

kellyi commented:
Note from https://github.com/open-apparel-registry/open-apparel-registry/pull/647#issuecomment-509779056

Also

- [ ] Make the website textbox be red when the URL is incorrect in the Claim application form
- [ ] Make the website clickable when reviewing a claim as an admin

jwalgran commented:
I will hold off on reviewing this until the migration and code conflicts are resolved.

kellyi commented:
Rebased this on `develop` and also made some adjustments

- we now use save rather than update for the claim update PUT endpoint so that things work with django-simple-history
- I made the website link clickable on the dashboard and also added the validation styling on the claim form website input

@jwalgran I believe this is ready for review!

jwalgran commented:
Looking at this now.

kellyi commented:
I pushed d67149dc1b112b4d868c3bbfff1a22a48a645f07 which is a new commit to adjust the facility type field options to match the new request on #634

kellyi commented:
Rebased on `develop`

kellyi commented on src/app/src/components/DashboardClaimsDetails.jsx:
```diff
@@ -147,18 +151,14 @@ function DashboardClaimsDetails({
 
                             if (!parentCompanyName) {
-                                return null;
+                                return '';
```
I believe this was to remedy a propTypes warning but is otherwise a noop.

kellyi commented on src/app/src/components/FacilityDetailSidebarClaimedInfo.jsx:
```diff
@@ -100,34 +118,30 @@ export default function FacilityDetailsSidebarClaimedInfo({
             />
             <ClaimInfoSection
-                value={facility.facility_average_lead_time}
+                value={facility.average_lead}
```
Yeah, that's right. It's called `average_lead` in the geojson properties but it was just silently absent here.

kellyi commented on src/django/api/models.py:
```diff
@@ -470,11 +470,18 @@ class FacilityClaim(models.Model):
-        help_text='The editable facility name for this claim.')
+        help_text='The editable official English facility name for the claim.',
+        verbose_name='facility name in English')
+    facility_name_non_english = models.CharField(
```
Sounds good. I'll update the field names throughout.

kellyi replied:
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/components/ClaimFacilityContactInfoStep.jsx:
```diff
@@ -52,6 +60,23 @@ function ClaimFacilityContactInfoStep({
+                    </Typography>
+                </InputLabel>
+                <TextField
+                    error={isEmpty(jobTitle)}
```
I fixed this in 4c3b188b86feb65c36d0ce6393d28827ffdd9a61

kellyi commented on src/django/api/views.py:
```diff
@@ -2029,44 +2033,115 @@ def get_claimed_details(self, request, pk=None):
+                .data \
+                .get('facility_average_lead_time')
+
+            claim_to_update.facility_workers_count = request \
```
Fixed this in 6449b9fcfbfeef54bcd8a04ec547bce95c3b6168 in two parts:

- prevent the client from sending non-int values for these fields
- if the api encounters non-int values here, they are ignored and the fields are set to None

kellyi commented on src/app/src/components/ClaimedFacilitiesDetails.jsx:
```diff
@@ -27,7 +27,10 @@ import COLOURS from '../util/COLOURS';
     clearClaimedFacilityDetails,
-    updateClaimedFacilityName,
+    updateClaimedFacilityNameEnglish,
+    updateClaimedFacilityNameNonEnglish,
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/components/ClaimedFacilitiesDetails.jsx:
```diff
@@ -312,9 +318,15 @@ function ClaimedFacilitiesDetails({
+                    disabled={updating}
+                />
+                <InputSection
+                    label="Facility name (non-English language)"
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/components/ClaimedFacilitiesDetails.jsx:
```diff
@@ -229,14 +232,17 @@ function ClaimedFacilitiesDetails({
     clearDetails,
-    updateFacilityName,
+    updateFacilityNameEnglish,
+    updateFacilityNameNonEnglish,
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/components/ClaimedFacilitiesDetails.jsx:
```diff
@@ -526,7 +574,10 @@ ClaimedFacilitiesDetails.propTypes = {
     clearDetails: func.isRequired,
-    updateFacilityName: func.isRequired,
+    updateFacilityNameEnglish: func.isRequired,
+    updateFacilityNameNonEnglish: func.isRequired,
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/components/ClaimedFacilitiesDetails.jsx:
```diff
@@ -595,7 +646,12 @@ function mapDispatchToProps(
+        updateFacilityNameEnglish: makeDispatchValueFn(
+            updateClaimedFacilityNameEnglish,
+        ),
+        updateFacilityNameNonEnglish: makeDispatchValueFn(
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/components/FacilityDetailSidebarClaimedInfo.jsx:
```diff
@@ -59,15 +59,17 @@ export default function FacilityDetailsSidebarClaimedInfo({
+                value={facility.name_english}
+            />
+            <ClaimInfoSection
+                label="Name (non-English language)"
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/reducers/ClaimedFacilityDetailsReducer.js:
```diff
@@ -9,7 +9,8 @@ import {
     completeUpdateClaimedFacilityDetails,
-    updateClaimedFacilityName,
+    updateClaimedFacilityNameEnglish,
+    updateClaimedFacilityNameNonEnglish,
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/reducers/ClaimedFacilityDetailsReducer.js:
```diff
@@ -79,12 +82,38 @@ export default createReducer({
+            facility_name_english: { $set: name },
+        },
+    }),
+    [updateClaimedFacilityNameNonEnglish]: (state, name) => update(state, {
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/app/src/util/propTypes.js:
```diff
@@ -259,14 +259,17 @@ export const facilityClaimPropType = shape({
     facility_description: string.isRequired,
-    facility_name: string.isRequired,
+    facility_name_english: string.isRequired,
+    facility_name_non_english: string.isRequired,
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/django/api/serializers.py:
```diff
@@ -440,7 +440,8 @@ def get_claim_info(self, facility):
                     'description': claim.facility_description,
-                    'name': claim.facility_name,
+                    'name_english': claim.facility_name_english,
+                    'name_non_english': claim.facility_name_non_english,
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/django/api/serializers.py:
```diff
@@ -571,12 +575,14 @@ class ApprovedFacilityClaimSerializer(ModelSerializer):
         model = FacilityClaim
-        fields = ('id', 'facility_description', 'facility_name',
+        fields = ('id', 'facility_description',
+                  'facility_name_english', 'facility_name_non_english',
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/django/api/views.py:
```diff
@@ -2035,7 +2035,10 @@ def get_claimed_details(self, request, pk=None):
-                facility_name=request.data.get('facility_name'),
+                facility_name_english=request.data
+                .get('facility_name_english'),
+                facility_name_non_english=request.data
```
Done in 65b7fa47ea0f260e54b8b90423646b8709f3507f

kellyi commented on src/django/api/views.py:
```diff
@@ -2040,53 +2040,108 @@ def get_claimed_details(self, request, pk=None):
-                office_country_code=request.data.get('office_country_code'),
-                office_phone_number=request.data.get('office_phone_number'),
-                office_info_publicly_visible=request.data
+            claim_to_update = FacilityClaim \
```
Good suggestion. I adjusted this in 


eaf286f 

and

6e649a0e792d46c45267c37ec1ac19c24d2f4fd1


The latter commit adjusts the order of how the non-dictionary settings get saved in order to ensure we save those fields too.

kellyi commented:
@jwalgran this is ready for another review!

jwalgran commented:
Looking at this now.

kellyi commented:
Commit message for cf92229249e6a6b1a53287c35399e82f62e5de7d
 is incorrect. the `isInteger || isInt` thing is to check whether the value is an int or can be cast to an int.

kellyi commented:
> Fixes look good and work well. Thanks. Mind the squash.

Awesome, thanks!
